core/src/main/java/com/netease/arctic/IcebergFileEntry.java

```diff
@@ -18,45 +18,51 @@
 
 package com.netease.arctic;
 
+import com.netease.arctic.utils.ManifestEntryFields;
 import org.apache.iceberg.ContentFile;
 
 /**
- * Entry of Iceberg ContentFile, include ContentFile, snapshotId and sequenceNumber.
- * The sequenceNumber is inherited from metadata, not the actual value in manifest file.
+ * Entry of Iceberg ContentFile, include status, ContentFile, snapshotId and sequenceNumber.
+ * If the actual sequenceNumber in manifest file is null, it will be inherited from metadata,
+ * and it may be null for DELETED entries with iceberg version < 1.0.0.
  */
 public class IcebergFileEntry {
-  private Long snapshotId;
-  private long sequenceNumber;
-  private ContentFile<?> file;
+  private final Long snapshotId;
+  private final Long sequenceNumber;
+  private final ManifestEntryFields.Status status;
+  private final ContentFile<?> file;
 
-  public IcebergFileEntry(Long snapshotId, long sequenceNumber, ContentFile<?> file) {
+  public IcebergFileEntry(Long snapshotId, Long sequenceNumber,
+                          ManifestEntryFields.Status status, ContentFile<?> file) {
     this.snapshotId = snapshotId;
     this.sequenceNumber = sequenceNumber;
+    this.status = status;
     this.file = file;
   }
 
   public Long getSnapshotId() {
     return snapshotId;
   }
 
-  public void setSnapshotId(Long snapshotId) {
-    this.snapshotId = snapshotId;
-  }
-
-  public long getSequenceNumber() {
+  public Long getSequenceNumber() {
     return sequenceNumber;
   }
 
-  public void setSequenceNumber(long sequenceNumber) {
-    this.sequenceNumber = sequenceNumber;
-  }
-
   public ContentFile<?> getFile() {
     return file;
   }
 
-  public void setFile(ContentFile<?> file) {
-    this.file = file;
+  public ManifestEntryFields.Status getStatus() {
+    return status;
+  }
+
+  @Override
+  public String toString() {
+    return "IcebergFileEntry{" +
```

Comment:
How about using `Objects.toStringHelpers` template to generate `toString` method as we are dependent on guava lib. DEA has already implemented this template, so you just need to select it when generating the toString method.

Reply:
OK, I refactor it.